src/combinatorics/simple_graph/strongly_regular.lean

```diff
@@ -0,0 +1,147 @@
+/-
+Copyright (c) 2021 Alena Gusakov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Alena Gusakov
+-/
+import combinatorics.simple_graph.basic
+/-!
+# Strongly regular graphs
+
+## Main definitions
+
+* `common_neighbors` is the intersection of the neighbor sets of two given vertices
+
+* `G.is_SRG_of n k l m` is a structure for a `simple_graph` with conditions
+  * The cardinality of the vertex set is `n`
+  * `G` is a regular graph with degree `k`
+  * The number of common neighbors between any two adjacent vertices in `G` is `l`
+  * The number of common neighbors between any two nonadjacent vertices in `G` is `m`
+
+## TODO
+- Define the complement of a graph
+- Prove that the complement of a strongly regular graph is strongly regular
+- Prove that the parameters of a strongly regular graph
+  obey relation `(n - k - 1) * m = k * (k - l - 1)`
+- Prove that if I is the identity matrix and J is the all-one matrix,
+  then the adj matrix A of SRG obeys relation A^2 = kI + lA + m(J - I - A)
+-/
+universes u
+
+namespace simple_graph
+variables {V : Type u}
+variables (G : simple_graph V)
+
+variables  [fintype V] [locally_finite G]
+
+/--
+A graph is strongly regular with parameters `n k l m` if
+ * its vertex set has cardinality `n`
+ * it is regular with degree `k`
+ * every pair of adjacent vertices has `l` common neighbors
+ * every pair of nonadjacent vertices has `m` common neighbors
+-/
+structure is_SRG_of (n k l m : ℕ) : Prop :=
+(card : fintype.card V = n)
+(regular : G.is_regular_of_degree k)
+(adj_common : ∀ (v w : V), G.adj v w → fintype.card (G.common_neighbors v w) = l)
+(nadj_common : ∀ (v w : V), ¬ G.adj v w → fintype.card (G.common_neighbors v w) = m)
+
+lemma degree_lt_card_verts (G : simple_graph V) (v : V) : G.degree v < fintype.card V :=
+begin
+  classical,
+  rw ← finset.card_univ,
+  rw degree,
+  apply finset.card_lt_card,
+  rw finset.ssubset_iff,
+  use v,
+  split,
+  { simp },
+  { apply finset.subset_univ },
+end
```

Comment:
This can go in simple_graph/basic.lean at the end of the `finite` section.

Compressed the proof:
```lean
lemma degree_lt_card_verts (G : simple_graph V) (v : V) : G.degree v < fintype.card V :=
begin
  classical,
  rw [degree, ←finset.card_univ],  -- (this line is actually not needed)
  apply finset.card_lt_card,
  rw finset.ssubset_iff,
  exact ⟨v, by simp, finset.subset_univ _⟩,
end
```